ci: gitlabform pipeline broken after gitlab version 18 release

The project's CI seems to be broken as multiple PRs have been failing to pass, preventing changes from being merged. For example: following dependency updates should not be failing. This started happening since the CI picked up latest version of gitlab, which happens to be v18 released by GitLab.

- #1026 
- #1027 
- #1028 

Snippet of the failure:

```
=========================== short test summary info ============================
FAILED tests/acceptance/standard/test_job_token_scope.py::TestProjectJobTokenScope::test__disable_limit_access_to_this_project - assert not True
 +  where True = <ProjectJobTokenScope>.inbound_enabled
======== 1 failed, 177 passed, 2 skipped, 3 rerun in 588.63s (0:09:48) =========
```

There have been breaking change introduced in GitLab v18 related to CI Job Token Scope. This might require refactor in GitLabForm's logic/test cases to reflect the change accordingly.

https://docs.gitlab.com/update/deprecations/?removal_milestone=18.0&breaking_only=true#cicd-job-token---authorized-groups-and-projects-allowlist-enforcement

https://docs.gitlab.com/update/deprecations/?removal_milestone=18.0&breaking_only=true#cicd-job-token---limit-access-from-your-project-setting-removal

Unfortunately cannot merge any new changes to this project until the CI is fixed.

--- a/docs/reference/job_token_scope.md
+++ b/docs/reference/job_token_scope.md
@@ -1,16 +1,29 @@
 # CI/CD job token scope
 
 ## Project CI/CD job token scope
-This section purpose is to manage the [project CI/CD job token scope](https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html#control-job-token-access-to-your-project)
+This section's purpose is to manage the [project CI/CD job token scope](https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html#control-job-token-access-to-your-project).
 
+The GitLab UI refers to this setting under "Authorized groups and projects". This controls which projects can use CI/CD job tokens to authenticate with the current project.
 
-The values are like documented at [Job Token Scope API docs](https://docs.gitlab.com/ee/api/project_job_token_scopes.html), **except the Enabled**.
+There are two main options in the UI:
+*   **All groups and projects**: This option disables any restrictions, allowing any project to use a job token to access this project. This corresponds to the API field `inbound_enabled` being `false`.
+*   **Only this project and any groups and projects in the allowlist**: This option restricts access to only the current project and those explicitly added to the allowlist. This corresponds to the API field `inbound_enabled` being `true`.
 
-We use `limit_access_to_this_project` as the variable name for restricting access to the Project from other projects, rather than `inbound_enabled` in the GET and `enabled` in the PATCH requests defined in the api, in line with GitLab's UI and [intended language](https://docs.gitlab.com/ee/update/deprecations.html#default-cicd-job-token-ci_job_token-scope-changed-1).
+It's important to note that this project-level setting can only be controlled if the instance-level setting `enforce_ci_inbound_job_token_scope_enabled` is disabled. This instance setting requires admin privileges to change.
 
-You can:
+In `gitlabform`, we use the `limit_access_to_this_project` setting to control the "Authorized groups and projects" option. Setting `limit_access_to_this_project: true` enables the restriction ("Only this project and any groups and projects in the allowlist"), while `limit_access_to_this_project: false` disables it ("All groups and projects").
 
-* [Limit access to a project](https://docs.gitlab.com/ee/api/project_job_token_scopes.html#patch-a-projects-cicd-job-token-access-settings)
+The [Job Token Scope API docs](https://docs.gitlab.com/ee/api/project_job_token_scopes.html) use `inbound_enabled` in GET requests and `enabled` in PATCH requests, which can be confusing. The `gitlabform` `limit_access_to_this_project` setting maps to the API's `enabled` field in PATCH requests.
+
+In addition to the main setting, you can manage the project's CI/CD job token allowlist. This allowlist determines which other projects and groups are authorized to use their job tokens to access this project when the "Only this project and any groups and projects in the allowlist" option is enabled.
+
+Within the `allowlist` section of the configuration, you can use the `enforce` setting:
+
+*   When `enforce: true` is set, `gitlabform` will ensure that the allowlist in GitLab exactly matches the `projects` and `groups` specified in your configuration. Any projects or groups found in the GitLab allowlist that are *not* listed in your `gitlabform` configuration will be removed.
+*   If `enforce` is not set or set to `false`, `gitlabform` will only add the projects and groups specified in the configuration to the allowlist, leaving any existing entries in GitLab untouched.
+
+You can add projects and groups to the allowlist using their ID or path/name:
+
 * [Add other projects to the job token allowlist](https://docs.gitlab.com/ee/api/project_job_token_scopes.html#add-a-project-to-a-cicd-job-token-inbound-allowlist)
 * [Add groups to the job token allowlist](https://docs.gitlab.com/ee/api/project_job_token_scopes.html#add-a-group-to-a-cicd-job-token-allowlist)
 

--- a/tests/acceptance/standard/test_job_token_scope.py
+++ b/tests/acceptance/standard/test_job_token_scope.py
@@ -35,16 +35,30 @@ class TestProjectJobTokenScope:
         project: Project,
     ):
         """
-        Test that project-level job token scope can be disabled when the instance setting allows it.
-        
-        This test requires:
-        1. GitLab instance setting 'enforce_ci_inbound_job_token_scope_enabled' to be configurable
-        2. Admin access to modify instance settings
+        Test that project-level 'Limit access to this project' setting can be configured
+        when the instance-level setting 'enforce_ci_inbound_job_token_scope_enabled' is disabled.
+
+        The test verifies that:
+        1. When instance setting is disabled, projects can control their own 'inbound_enabled' setting
+        2. The gitlabform config 'limit_access_to_this_project: false' correctly sets
+           the project's 'inbound_enabled' to false via the GitLab API
+
+        This test requires admin access to modify the instance setting.
         """
-        # Setup: Ensure instance setting allows project-level configuration
+        # Setup: Disable instance-level enforcement to allow project-level configuration
         instance_settings = gl.settings.get()
         instance_settings.enforce_ci_inbound_job_token_scope_enabled = False
-        instance_settings.save()
+        
+        # Retry settings update in case of temporary resource issues
+        max_retries = 3
+        for attempt in range(max_retries):
+            try:
+                instance_settings.save()
+                break
+            except Exception as e:
+                if attempt == max_retries - 1:
+                    raise Exception(f"Failed to update instance settings after {max_retries} attempts: {str(e)}")
+                print(f"Attempt {attempt + 1} failed, retrying...")
 
         try:
             # Setup: Enable limit access at project level
@@ -65,9 +79,12 @@ class TestProjectJobTokenScope:
 
         finally:
             # Cleanup: Restore instance setting to its original state
-            instance_settings = gl.settings.get()
-            instance_settings.enforce_ci_inbound_job_token_scope_enabled = True
-            instance_settings.save()
+            try:
+                instance_settings = gl.settings.get()
+                instance_settings.enforce_ci_inbound_job_token_scope_enabled = True
+                instance_settings.save()
+            except Exception as e:
+                print(f"Warning: Failed to restore instance settings: {str(e)}")
 
     def test__add_other_project_to_job_token_scope_by_name(
         self,

--- a/tests/acceptance/standard/test_job_token_scope.py
+++ b/tests/acceptance/standard/test_job_token_scope.py
@@ -31,25 +31,43 @@ class TestProjectJobTokenScope:
 
     def test__disable_limit_access_to_this_project(
         self,
+        gl,
         project: Project,
     ):
-        self._setup_limit_access_state(project, True)
-
-        job_token_scope = f"""
-        projects_and_groups:
-          {project.path_with_namespace}:
-            job_token_scope:
-              limit_access_to_this_project: false
         """
+        Test that project-level job token scope can be disabled when the instance setting allows it.
+        
+        This test requires:
+        1. GitLab instance setting 'enforce_ci_inbound_job_token_scope_enabled' to be configurable
+        2. Admin access to modify instance settings
+        """
+        # Setup: Ensure instance setting allows project-level configuration
+        instance_settings = gl.settings.get()
+        instance_settings.enforce_ci_inbound_job_token_scope_enabled = False
+        instance_settings.save()
 
-        run_gitlabform(job_token_scope, project)
+        try:
+            # Setup: Enable limit access at project level
+            self._setup_limit_access_state(project, True)
 
-        scope = project.job_token_scope.get()
+            # Test: Disable limit access through gitlabform
+            config = f"""
+            projects_and_groups:
+              {project.path_with_namespace}:
+                job_token_scope:
+                  limit_access_to_this_project: false
+            """
+            run_gitlabform(config, project)
 
-        # inbound_enabled is what GL returns from GET
-        # https://docs.gitlab.com/ee/api/project_job_token_scopes.html#get-a-projects-cicd-job-token-access-settings
-        # to denote whether "Limit Access to this Project" is enabled or not
-        assert not scope.inbound_enabled
+            # Verify: Check that limit access is disabled
+            scope = project.job_token_scope.get()
+            assert not scope.inbound_enabled, "Project should have limit access disabled"
+
+        finally:
+            # Cleanup: Restore instance setting to its original state
+            instance_settings = gl.settings.get()
+            instance_settings.enforce_ci_inbound_job_token_scope_enabled = True
+            instance_settings.save()
 
     def test__add_other_project_to_job_token_scope_by_name(
         self,

--- a/tests/acceptance/standard/test_job_token_scope.py
+++ b/tests/acceptance/standard/test_job_token_scope.py
@@ -48,17 +48,7 @@ class TestProjectJobTokenScope:
         # Setup: Disable instance-level enforcement to allow project-level configuration
         instance_settings = gl.settings.get()
         instance_settings.enforce_ci_inbound_job_token_scope_enabled = False
-
-        # Retry settings update in case of temporary resource issues
-        max_retries = 3
-        for attempt in range(max_retries):
-            try:
-                instance_settings.save()
-                break
-            except Exception as e:
-                if attempt == max_retries - 1:
-                    raise Exception(f"Failed to update instance settings after {max_retries} attempts: {str(e)}")
-                print(f"Attempt {attempt + 1} failed, retrying...")
+        instance_settings.save()
 
         try:
             # Setup: Enable limit access at project level

--- a/tests/acceptance/standard/test_job_token_scope.py
+++ b/tests/acceptance/standard/test_job_token_scope.py
@@ -48,7 +48,7 @@ class TestProjectJobTokenScope:
         # Setup: Disable instance-level enforcement to allow project-level configuration
         instance_settings = gl.settings.get()
         instance_settings.enforce_ci_inbound_job_token_scope_enabled = False
-        
+
         # Retry settings update in case of temporary resource issues
         max_retries = 3
         for attempt in range(max_retries):

